test/unit/test_sap_http_oauth.py

```
@@ -0,0 +1,295 @@
+        )
+
+        mock_get_token.assert_called_once_with(
+            'https://auth.example.com', 'my-client-id', 'my-client-secret',
+            user='user@sap.com', password='secret'
+        )
+        self.assertIsInstance(client._auth, BearerAuth)
+
+    @patch('sap.http.oauth.get_token')
+    def test_uses_basic_auth_when_no_token_url(self, mock_get_token):
+        from requests.auth import HTTPBasicAuth
+        client = HTTPClient(host='c50.example.com', user='ELBEZI', password='pass')
```

Ruff (0.15.12)

[error] Possible hardcoded password assigned to argument: "password"

(S106)